util/file.py

```diff
@@ -0,0 +1,43 @@
+import json
+import os
+import random
+import pyminizip
+
+from typing import Any, Dict, Optional, Tuple
+from zipfile import ZipFile
+from exceptions.exceptions import BadPasswordError
+
+
+def zip_to_json(local_file_name: str, key: Optional[str] = None) -> Dict[str, Any]:
+    zip_file = ZipFile(local_file_name)
+    file_name = zip_file.namelist()[0]
+    if key:
+        key = key.encode()
+        zip_file.setpassword(key)
+    
+    try:
+        data = json.loads(zip_file.read(file_name, key).decode())
+        return data
+    except RuntimeError as error:
+        if "password" in str(error):
+            raise BadPasswordError
+        
+def zip_to_json_file(zip_file_path: str, key: Optional[str] = None) -> str:
+    json_data = zip_to_json(zip_file_path, key)
+    file_name = f"{zip_file_path}{random.randint(1,999)}.json" 
```

Comment:
why are we using a random integer in the file name? Usually i'd expect a while loop that checks existing and return the first free filename

Reply:
I mean, both work fine, but of course I can rewrite it

Reply:
since we don't check the file name this would (of course only in rare instances) create an issue with the same file name being used for two different tasks. to be honest with our current size I don't expect this to ever happen. However, this is not a solution that works 100% of the time and might result in very hard-to-find (and debug) issues.

I would be fine-er with it if it's wrapped in a loop to check the file name.

To solve this for future use as well i'd create a small function like `get_free_file_name(path:str,file_name:str, increment:int = 1):`

Reply:
I agree, your way is more fail safe, will quickly rewrite it